### DISCUSSION TITLE
Entrega-1

### DISCUSSION THREAD

**Discussion by lKubi** [lKubi/AEDI-PROJECT#3] [General]
### Buenas noches xd,

Esto es rollo Reddit, así que es muy intuitivo usarlo.

Necesito que respondan a esto una vez hayan revisado el código. Y si tienen alguna pregunta sobre él la pongan aquí.

También una vez hayan revisado el código y si quieren cambiar algo o añadir algo lo pongan aquí para tener más control de eso y no se pierda por el grupo de WhatsApp.

Hasta que todos hayan respondido aquí (Entendiendo que revisaron el código, lo entienden y no quieran añadir ninguna función más) no se moverán los cambios a la rama _main_ cerrando así la rama _Entrega-1_. Y una vez los cambios se encuentren en la rama _main_ es cuando se entregará el **.zip** con la primera entrega al Moovi.

Cualquier cosa ya saben que no duden en preguntar que intentaré explicarlo lo mejor que pueda y ayudarles.

:)

@Alestar12 @Donakyoin @xMaurox10 @Migui-A @Mtouri.

**Comment by Migui-A** [comment]
Mirando todas las clases del proyecto y los métodos implementados lo único que me lia un poco es utilizar las clases collection (ya que era la primera vez que escuchaba de ellas) pero quitando eso, el resto pienso que es muy intuitivo de aprender y con los comentarios no te pierdes... las clase jugador lo único a manejar es la lista mano de eliminar o insertar peor eso se dio en clase... yla clase carta no hay ciencia de perderse, la clase IU es más o menos lo que estuvimos haciendo en PROII de pedir datos verificando unas condiciones y retornando solo que hay que hacerlo con collection... la clase Baraja lo único nuevo es el método shuffle para barajar y por último la clase juego pedir los nombres de los jugadores con la clase pedirDatosJugadores irlos creando y con el método ramdom inicializar la partida... por último decir que solo por capricho quitaría esos métodos de mostrarJugadores ya que si no los utilizamos (puede que a futuro los utilizaremos pero lo dudo) de que sirven que estén ahí... por el resto no lo tocaria nada... creo que me excedí con las palabras pero es lo que pienso del proyecto que hicistes Luis ñ, espero que esto ayude al menos y si no me avisas e intentamos arreglarlo xd

**Comment by Migui-A** [comment]
Viendo que estaba mirando versión desactualizada xd los métodos de mostrar en IU es casi como un toString() (desde mi punto de vista) sin complicarnos la vida por el resto sigo diciendo que no toco nada de nada

**Comment by Mtouri** [comment]
Revisando el proyecto, se entiende perfectamente y yo no cambiaría nada por el momento. Sin embargo, se me ocurre añadir algo en la clase IU: a la hora de revisar que los nombres de los jugadores introducidos no estén vacíos, también podríamos comprobar que no sean solo números. La solución podría ser algo como esto: https://pastebin.com/VLGTqiBQ

Entiendo que esto en el enunciado no se pide explícitamente y es una tontería pero entiendo que los jugadores no deberían poder ser números, así que si todos están de acuerdo, podemos añadirlo. 


Mañana le doy otra vuelta y a ver si se me ocurre alguna otra cosa, pero por ahora, esta perfe.

**Comment by maurozelenka** [comment]
Sin duda uno de los mejores proyectos que he visto en la vida. Yo no tocaría nada así esta perfecto. Siento no poder ayudado mucho pero en la segunda entrega estaré dispuesto a dejar mis cualidades en este proyecto tan importante para nosotros. 

Un saludo

**Comment by DonakyoiN** [comment]
Bueno, dándole otra revisada más al código, como mencioné en el Whatsapp el código está excelente, se comprende bien los métodos utilizados y los comentarios realizados dentro del código te facilitan entenderlo. El único detallito lo que mencioné, modificaría la lectura del try catch, cambiándolo de System.out a System.err, es una tontería pero creo que así queda más de acorde a la excepción. Por el resto está todo perfecto y lo mantendría como está

**Comment by lKubi** [comment]
HIJOS DE PUTA PODÍAN RESPONDER ANTES GRACIAS.

CON CARIÑO, TQM :) <3
